File: interlayer/src/main/java/com/street/interlayer/model/Transaction.java
package com.street.interlayer.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.Date;

@Data
public class Transaction {

    private String id;
    @JsonProperty("author")
    private Long authorId;
    private Account account;
    private Double amount;
    private Date createdAt;
    @JsonInclude(JsonInclude.Include.NON_NULL)
    private String recipientWallet;

}

File: micro/src/main/kotlin/com/street/micro/entity/Transaction.kt
package com.street.micro.entity

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer
import com.mongodb.lang.Nullable
import org.bson.types.ObjectId
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDateTime

@Document
data class Transaction(
    @Id
    @JsonSerialize(using = ToStringSerializer::class)
    val id: ObjectId = ObjectId.get(),
    val author: Long,
    val amount: Double,
    val created_at: LocalDateTime = LocalDateTime.now(),
    val recipientWallet: ObjectId?
)
